--- drivers/net/wireless/mwifiex/main.c	2013-11-30 22:31:45.000000000 +0100
+++ drivers/net/wireless/mwifiex/main.c	2013-12-08 22:35:48.250493936 +0100
@@ -477,12 +477,20 @@
 	/* Create station interface by default */
 	wdev = mwifiex_add_virtual_intf(adapter->wiphy, "mlan%d",
 					NL80211_IFTYPE_STATION, NULL, NULL);
-	if (IS_ERR(wdev)) {
+        if (IS_ERR(wdev)) {
 		dev_err(adapter->dev, "cannot create default STA interface\n");
 		rtnl_unlock();
 		goto err_add_intf;
 	}
-	rtnl_unlock();
+	
+	/* Create AP interface by default */
+	if (!mwifiex_add_virtual_intf(adapter->wiphy, "uap%d",
+                                     NL80211_IFTYPE_AP, NULL, NULL)) {
+               dev_err(adapter->dev, "cannot create default AP interface\n");
+               goto err_add_intf;
+	}
+   
+        rtnl_unlock();
 
 	mwifiex_drv_get_driver_version(adapter, fmt, sizeof(fmt) - 1);
 	dev_notice(adapter->dev, "driver_version = %s\n", fmt);
